src/index.js

```diff
@@ -48,49 +37,79 @@ class Notification {
    */
   send(event, notification) {
     const user = notification.to;
-    this.io.sockets[user].emit(event, notification, (ack) => {
-      if (ack) {
-        notificationController
-        .update(notification._id, { sent: true }, //eslint-disable-line
-          (err) => {
-            if (err) throw err;
-          });
-      }
+    const notificationUpdate = new Promise((resolve, reject) => {
+      this.io.sockets[user].emit(event, notification, (ack) => {
+        if (ack) {
+          this.notificationController
+            .update(notification._id, { sent: true }) //eslint-disable-line
+            .then(updatedNotification => resolve(updatedNotification))
+            .catch((err) => {
+              reject(err);
+            });
+        } else {
+          resolve();
+        }
+      });
     });
+    return notificationUpdate;
   }
 
   /**
    * Sends all the unread notifications by user id
    * @param  {string} userId user identifier
    */
   sendUnread(userId) {
-    const self = this;
     if (userId) {
-      notificationController
-      .getUnread(userId, (err, data) => {
-        if (err) throw err;
-        data.forEach((notification) => {
-          self.send('unread', notification);
-        });
+      const notificatonsSent = new Promise((resolve, reject) => {
+        this.notificationController
+          .getAllUnread(userId)
+          .then((notifications) => {
+            const sentNotifications = notifications.map((notification) => {
+              const sentNotification = this.send('unread', notification);
+              return sentNotification;
+            });
+            return Promise.all(sentNotifications);
+          })
+          .then((notifications) => {
+            resolve(notifications);
+          })
+          .catch((err) => {
+            reject(err);
+          });
       });
+      return notificatonsSent;
     }
+
+    return null;
   }
 
   /**
    * Sends all unsent notifications by user id-
    * @param  {string} userId User identifier
    */
   sendUnsent(userId) {
-    const self = this;
     if (userId) {
-      notificationController
-      .getUnsent(userId, (err, data) => {
-        if (err) throw err;
-        data.forEach((notification) => {
-          self.send('news', notification);
-        });
+      const notificatonsSent = new Promise((resolve, reject) => {
+        this.notificationController
+          .getAllUnsent(userId)
+          .then((notifications) => {
+            const sentNotifications = notifications.map((notification) => {
+              const sentNotification = this.send('news', notification);
```

Comment:
Unrelated, but you used events for this right? I think we should be consistent, if we are using promises we should use promises everywhere (even for tests)